### ISSUE TITLE
Biopython fails to parse/read gene db xml files.

### ISSUE THREAD

**Issue by tejaswirohit** [biopython/biopython#1571]
### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

```python
import sys; print(sys.version)
import platform; print(platform.python_implementation()); print(platform.platform())
import Bio; print(Bio.__version__)
```

(*Please copy and run the above in your Python, and copy-and-paste the output*)
```
3.6.2 (default, Sep 15 2017, 07:44:18) 
Darwin-17.4.0-x86_64-i386-64bit
1.70
```

### Expected behaviour

```python
records = Entrez.read(handle)
```

### Actual behaviour

```
Traceback (most recent call last):
  File "biotest.py", line 10, in <module>
    records = Entrez.read(handle)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/Bio/Entrez/__init__.py", line 453, in read
    record = handler.read(handle)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/Bio/Entrez/Parser.py", line 233, in read
    self.parser.ParseFile(handle)
  File "/private/tmp/python3-20170915-3144-15ynttz/Python-3.6.2/Modules/pyexpat.c", line 414, in StartElement
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/Bio/Entrez/Parser.py", line 368, in startElementHandler
    raise ValidationError(name)
Bio.Entrez.Parser.ValidationError: Failed to find tag 'Gene-commentary_rna' in the DTD. To skip all tags that are not represented in the DTD, please call Bio.Entrez.read or Bio.Entrez.parse with validate=False.
```

### Steps to reproduce
```python
ids = ['112038883']
handle = Entrez.efetch(db='gene', retmode='xml', id=','.join(ids))
records = Entrez.read(handle)
```

Whe I do try to disable validation by passing Entrez.read(handle, validate=False) I get:
```
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/Bio/Entrez/Parser.py", line 379, in startElementHandler
    current.append(object)
AttributeError: 'str' object has no attribute 'append'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "biotest.py", line 10, in <module>
    records = Entrez.read(handle, validate=False)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/Bio/Entrez/__init__.py", line 453, in read
    record = handler.read(handle)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/Bio/Entrez/Parser.py", line 233, in read
    self.parser.ParseFile(handle)
  File "/private/tmp/python3-20170915-3144-15ynttz/Python-3.6.2/Modules/pyexpat.c", line 414, in StartElement
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/Bio/Entrez/Parser.py", line 381, in startElementHandler
    current[name] = object
TypeError: 'str' object does not support item assignment
```

Not sure how to debug this or get past it.

**Comment by peterjc** [comment]
Very similar to #1575 - it looks like the NCBI has got their XML and DTD out of sync again.

You may be better of using one of the Python standard library XML parsers instead, e.g. ElementTree (which is what our parser uses internally).

**Comment by mdehoon** [comment]
Fixed by pull request  #1644 .
